cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -110,84 +110,123 @@ class CircuitOperation(ops.Operation):
             with repetitions or repetition_ids.
     """
 
-    _hash: Optional[int] = dataclasses.field(default=None, init=False)
-    _cached_measurement_key_objs: Optional[AbstractSet['cirq.MeasurementKey']] = dataclasses.field(
-        default=None, init=False
-    )
-    _cached_control_keys: Optional[AbstractSet['cirq.MeasurementKey']] = dataclasses.field(
-        default=None, init=False
-    )
-    _cached_mapped_single_loop: Optional['cirq.Circuit'] = dataclasses.field(
-        default=None, init=False
-    )
-
-    circuit: 'cirq.FrozenCircuit'
-    repetitions: IntParam = 1
-    qubit_map: Dict['cirq.Qid', 'cirq.Qid'] = dataclasses.field(default_factory=dict)
-    measurement_key_map: Dict[str, str] = dataclasses.field(default_factory=dict)
-    param_resolver: study.ParamResolver = study.ParamResolver()
-    repetition_ids: Optional[List[str]] = dataclasses.field(default=None)
-    parent_path: Tuple[str, ...] = dataclasses.field(default_factory=tuple)
-    extern_keys: FrozenSet['cirq.MeasurementKey'] = dataclasses.field(default_factory=frozenset)
-    use_repetition_ids: bool = True
-    repeat_until: Optional['cirq.Condition'] = dataclasses.field(default=None)
-
-    def __post_init__(self):
-        if not isinstance(self.circuit, circuits.FrozenCircuit):
-            raise TypeError(f'Expected circuit of type FrozenCircuit, got: {type(self.circuit)!r}')
+    def __init__(
+        self,
+        circuit: 'cirq.FrozenCircuit',
+        repetitions: int = 1,
+        qubit_map: Optional[Dict['cirq.Qid', 'cirq.Qid']] = None,
+        measurement_key_map: Optional[Dict[str, str]] = None,
+        param_resolver: Optional[study.ParamResolverOrSimilarType] = None,
+        repetition_ids: Optional[Sequence[str]] = None,
+        parent_path: Tuple[str, ...] = (),
+        extern_keys: FrozenSet['cirq.MeasurementKey'] = frozenset(),
+        use_repetition_ids: bool = True,
+        repeat_until: Optional['cirq.Condition'] = None,
+    ):
```

Comment:
Should the docstring be moved from the class to here?

Reply:
Done.